test/distributed/checkpoint/test_fsspec.py

```diff
@@ -176,6 +177,25 @@ def opt_at(opt, idx):
             opt_at(optim, 0)["exp_avg_sq"], opt_at(optim_2, 0)["exp_avg_sq"]
         )
 
+    @with_comms(init_rpc=False)
+    @skip_if_lt_x_gpu(2)
+    @requires_nccl()
+    @with_temp_dir
+    def test_overwrite(self):
+        t1, t2 = torch.randn(10), torch.randn(10)
+        dcp.save({"random": t1}, checkpoint_id=self.temp_dir)
+        dcp.save({"random": t2}, checkpoint_id=self.temp_dir)
+
+        sd = {"random": torch.zeros(10)}
+        dcp.load(sd, checkpoint_id=self.temp_dir)
+
+        self.assertTrue(torch.allclose(sd["random"], t2))
+
+        dcp.save(
+            {"random": t2}, storage_writer=FsspecWriter(self.temp_dir, overwrite=True)
```

Comment:
Should this be overwrite=False when exception is raised, As the default is overwrite=True.

Reply:
yes, and the format for assert raises is wrong as well, I'll have to take a look

Reply:
fixed, thanks @wz337 !